rtcpeerconnection.go

```diff
@@ -956,31 +955,46 @@ func (pc *RTCPeerConnection) NewRTCTrack(payloadType uint8, id, label string) (*
 		return nil, errors.New("codec payloader not set")
 	}
 
-	buf := make([]byte, 4)
-	_, err = rand.Read(buf)
-	if err != nil {
-		return nil, errors.New("failed to generate random value")
-	}
-
 	trackInput := make(chan media.RTCSample, 15) // Is the buffering needed?
-	ssrc := binary.LittleEndian.Uint32(buf)
-	go func() {
-		packetizer := rtp.NewPacketizer(
-			1400,
-			payloadType,
-			ssrc,
-			codec.Payloader,
-			rtp.NewRandomSequencer(),
-			codec.ClockRate,
-		)
-		for {
-			in := <-trackInput
-			packets := packetizer.Packetize(in.Data, in.Samples)
-			for _, p := range packets {
+	rawPackets := make(chan *rtp.Packet)
+	if ssrc == 0 {
+		buf := make([]byte, 4)
+		_, err = rand.Read(buf)
+		if err != nil {
+			return nil, errors.New("failed to generate random value")
+		}
+		ssrc = binary.LittleEndian.Uint32(buf)
+
+		go func() {
+			packetizer := rtp.NewPacketizer(
+				1400,
+				payloadType,
+				ssrc,
+				codec.Payloader,
+				rtp.NewRandomSequencer(),
+				codec.ClockRate,
+			)
+
+			for {
+				in := <-trackInput
+				packets := packetizer.Packetize(in.Data, in.Samples)
+				for _, p := range packets {
+					pc.networkManager.SendRTP(p)
+				}
+			}
+		}()
+		close(rawPackets)
+	} else {
+		// If SSRC is not 0, then we are working with an established RTP stream
+		// and need to accept raw RTP packets for forwarding.
+		go func() {
+			for {
+				p := <-rawPackets
```

Comment:
Can/should we copy the packets here?

It may confuse users if they try to send the same packet twice, it will become corrupted (because later we encrypt in place)

Reply:
Yeah, this bit me. On the one hand, it would certainly be more convenient for users to not have to deal with it. On the other hand, if we put the code for copying here, it will always impose the copying overhead even when it's not necessary. One way to address this might be to create an SFU implementation which handles this efficiently (c.f. [http/server.go](https://golang.org/src/net/http/server.go#L800)) and explicitly.

Reply:
Mind adding a comment to the godoc about needing to copy? Beyond that everything LGTM!